torchvision/transforms/functional_pil.py

```diff
@@ -308,28 +357,28 @@ def to_grayscale(img, num_output_channels):
 
 
 @torch.jit.unused
-def invert(img):
+def invert(img: Image.Image) -> Image.Image:
     if not _is_pil_image(img):
         raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
     return ImageOps.invert(img)
 
 
 @torch.jit.unused
-def posterize(img, bits):
+def posterize(img: Image.Image, bits: int) -> Image.Image:
     if not _is_pil_image(img):
         raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
     return ImageOps.posterize(img, bits)
 
 
 @torch.jit.unused
-def solarize(img, threshold):
+def solarize(img: Image.Image, threshold: int) -> Image.Image:
```

Comment:
Please verify. I think it's int. (after digging into PIL code)

Reply:
[Seems to be good](https://pillow.readthedocs.io/en/stable/reference/ImageOps.html?highlight=posterize#PIL.ImageOps.solarize)

> threshold – All pixels above this greyscale level are inverted.

Although I'm not sure if `float` would also be possible for floating point images (mode `F`).